### ISSUE TITLE
[WAITING ON PAYMENT] There is some unnecessary loading for few secs after workspace deletion - reported by @Santhosh-Sellavel 

### ISSUE THREAD

**Issue by mvtglobally** [Expensify/App#8035]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Create a workspace if you dont have already, using :heavy_plus_sign:  button.
2. Now navigate to workspace, click more button.
3. Click Delete workspace.
4. Confirm Delete
5. Now you see a growl saying Workspace Deleted
6. But there is some loading after the message.

## Expected Result:
Close immediately after delete confirmation

## Actual Result:
Stuck with loading for few secs

## Workaround:
unknown 

## Platform:

Where is this issue occurring?

- Web
- iOS
- Android
- Desktop App
- Mobile Web

**Version Number:** 1.1.41-0
**Reproducible in staging?:** Y 
**Reproducible in production?:** Y 
**Email or phone of affected tester (no customers):** 
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation 

https://user-images.githubusercontent.com/43995119/157186066-ea41b9d7-887c-4343-9eb1-7e1e40eccc38.mov

**Expensify/Expensify Issue URL:** 
**Issue reported by:** @Santhosh-Sellavel 
**Slack conversation:** https://expensify.slack.com/archives/C01GTK53T8Q/p1645813470191519

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @MitchExpensify (`AutoAssignerTriage`), see https://stackoverflow.com/c/expensify/questions/4749 for more details.

**Comment by MitchExpensify** [comment]
Nice bit of polish! Confirmed this is reproducible and issue is clear, labeling eng

**Comment by MelvinBot** [comment]
Triggered auto assignment to @MonilBhavsar (`Engineering`), see https://stackoverflow.com/c/expensify/questions/4319 for more details.

**Comment by MonilBhavsar** [comment]
Looks like this is happening as we're fetching reports after deleting a workspace. Should we only fetch if we are successful in deleting a policy? The idea is to show deleted chatrooms associated with policy, right?

https://github.com/Expensify/App/blob/c26e41573c6b6ea91abef85afc173ba6ecc4cd2c/src/libs/actions/Policy.js#L184

@neil-marcellini thoughts?

**Comment by neil-marcellini** [comment]
To fix the loading delay when deleting a workspace is successful, we should do an optimistic update, and assume that all rooms in the workspace will be marked as deleted. Then, in the background, we should fetch all reports / rooms again and make sure that the response is successful. If the response fails, we should notify the user that we were unable to delete their workspace and we should restore the rooms / reports to the ones that were fetched, which will show that the rooms were not deleted. [Here](https://expensify.slack.com/archives/C01GTK53T8Q/p1646675428796039) is the latest discussion on optimistic updates. Does that sound good? I think it will be better for the user than my initial implementation.

**Comment by MonilBhavsar** [comment]
Not overdue, coming back

**Comment by MonilBhavsar** [comment]
Coming back next week

**Comment by MelvinBot** [comment]
@MonilBhavsar Uh oh! This issue is overdue by 2 days. Don't forget to update your issues!

**Comment by MonilBhavsar** [comment]
Getting back this week

**Comment by MonilBhavsar** [comment]
Looking today

**Comment by MonilBhavsar** [comment]
On hold for API redesign

**Comment by MonilBhavsar** [comment]
Same ^

**Comment by MonilBhavsar** [comment]
On hold for API redesign

**Comment by MonilBhavsar** [comment]
Will be looking at this after reviewing doc, probably next week

**Comment by MonilBhavsar** [comment]
Working this week

**Comment by MonilBhavsar** [comment]
Closing this as it will be fixed by API refactoring. Feel free to reopen if issue persists.

**Comment by jliexpensify** [comment]
Cool, thanks @MonilBhavsar - going to pay @Santhosh-Sellavel the reporting fee of $250.

**Comment by jliexpensify** [comment]
Invited @Santhosh-Sellavel 

Internal - https://www.upwork.com/ab/applicants/1536307494591885312/job-details
External - https://www.upwork.com/jobs/~01e8f5d7c440d5a1a4

**Comment by MonilBhavsar** [comment]
Thanks @jliexpensify! There's discussion going on [this](https://github.com/Expensify/App/issues/8455) similar issue regarding payment. Should we hold this for now?

**Comment by jliexpensify** [comment]
Ah, good catch @MonilBhavsar - @Santhosh-Sellavel, the consensus in the other issue is:

 > to be eligible for the reporting bonus, the fix needs to be deployed to production.
 
 I'm happy to hold (I've just invited Santhosh atm), do we have an ETA for the fix?

**Comment by Santhosh-Sellavel** [comment]
Applied for job @jliexpensify

**Comment by jliexpensify** [comment]
@Santhosh-Sellavel - it looks like we'll pay once the issue is fixed, which should hopefully be in the coming weeks!

**Comment by Santhosh-Sellavel** [comment]
Not a problem, @jliexpensify!

**Comment by jliexpensify** [comment]
@MonilBhavsar - Hello! Any updates on this issue?

**Comment by MonilBhavsar** [comment]
Should be fixed by this refactor https://github.com/Expensify/Expensify/issues/215184

**Comment by jliexpensify** [comment]
Hi @MonilBhavsar - any updates here?

**Comment by MonilBhavsar** [comment]
Looks like the issue is fixed!
Feel free to reopen if issue persists.

**Comment by jliexpensify** [comment]
Great, going to pay @Santhosh-Sellavel the $250 now.

**Comment by jliexpensify** [comment]
@Santhosh-Sellavel - I had to re-create the job since it closed due to inactivity.

Internal -  https://www.upwork.com/ab/applicants/1577405728492056576/job-details
External - https://www.upwork.com/jobs/~01db0184c53dc2c418

I've hired you for payment.

**Comment by jliexpensify** [comment]
@Santhosh-Sellavel paid, cheers! Job closed.
